### PR TITLE
feat: Int32 support added (#103)

### LINKED ISSUE #103: Add types.Int32
It's nice to have explicit support fo Avro `int` type for compatibility with existing software

```
class Int32(int):
    ...
```

### PR DESCRIPTION
**Pull request by eugene-bright**
Closes #103
Relates #102

### CODE REVIEW

**Comment by marcosschroh** [comment]
Hi @eugene-bright 

Thanks for the `PR`. Looking good, but I do not understand what is the difference with the `int` type. Why do we need to define a new type that is working in the same same exact way as the `native type int`?

**Comment by eugene-bright** [comment]
It's a provisional support for Avro type int. It can be useful if compatibility with existing third party project is required.
Currently Python int compiles to Avro long.

**Comment by marcosschroh** [comment]
> It's a provisional support for Avro type int. It can be useful if compatibility with existing third party project is required.
> Currently Python int compiles to Avro long.

Could you give some examples of use cases?

**Comment by eugene-bright** [comment]
Example use case.
Existing application uses schema, containing Avro int value (without logical type). For convenience schema rewritten to dataclass-avroschema, but generated message must be correctly parsed by existing third party application. The types.Int32 must be used to enforce interoperability.

**Comment by marcosschroh** [comment]
After looking at it, now I see what you want to achieve. Sometimes without a description is a bit hard to understand.

So, now we have:

| avro type| python type|
|----------|------------|
|  long | int | 

and you want 

| avro type| python type|
|----------|------------|
|  long | int | 
|  int | types.Int32 |

Can you add tests and update the documentation?

File: dataclasses_avroschema/types.py [hunk 1]
```diff
@@ -70,4 +70,17 @@ def __repr__(self) -> str:
...
 
 
-CUSTOM_TYPES = ("Fixed", "Enum", "Decimal")
+@dataclasses.dataclass
+class Int32(int):
+    """
+    Represents an Avro int type
+    """
+
+    default: typing.Any = dataclasses.field(default=MissingSentinel)
```
Comment (marcosschroh) [inline dataclasses_avroschema/types.py:79]: This is not necessary. Just with this should be enough 

```python
@dataclasses.dataclass
class Int32(int):
    """
    Represents an Avro int type INT
    """
    pass
```

**Comment by eugene-bright** [comment]
I will work on tests and documentation on the next weekends

**Comment by judahrand** [comment]
> Thanks for the `PR`. Looking good, but I do not understand what is the difference with the `int` type. Why do we need to define a new type that is working in the same same exact way as the `native type int`?

I have an example of where this would be useful for me. I'm investigating using this library to define some event classes which will be serialised to Avro before being sent to BigQuery. For storage purposes it would be good to be able to specify the size of the `int` and `float`.

**Comment by eugene-bright** [comment]
Sorry, I was distracted from this work. I will try to resume it this weekend.

**Comment by judahrand** [comment]
If it's easy (probably is) would you mind copy-pasta-ing the work to also support `float`? I can also open a PR but since it's all already there might be easier for you?

**Comment by eugene-bright** [comment]
Ok, but you will need to wait a little

**Comment by judahrand** [comment]
I ended up having a moment to look at this so opened a competing implementation 👅

**Pull request closed without merging**